src/lstm.js

```diff
@@ -97,10 +100,30 @@ export function lstm(input, weight, recurrentWeight, steps, hiddenSize,
     cellState = nextCell;
 
     if (returnSequence) {
-      nextHidden = reshape(nextHidden, [1, numDirections, null, hiddenSize]);
+      // Expand [numDirections, batchSize, hiddenSize] to
+      // [steps, numDirections, batchSize, hiddenSize]
+      nextHidden = reshape(nextHidden, [1, numDirections, batchSize, hiddenSize]);
+      // Concat output sequence along axis 0 (steps dimension)
       sequence = (sequence ? concat([sequence, nextHidden], 0) : nextHidden);
```

Comment:
Could we just reverse the concatenation order and fix this issue?
```suggestion
      sequence = (sequence ? (backward ? concat([nextHidden, sequence], 0) : concat([sequence, nextHidden], 0)) : nextHidden);
```

You may want to declare `backward` as `let backward = dir == 1 || direction == 'backward';`

Reply:
Thanks for more consideration!

This proposal solution can only fix for 'backward' direction.
Here this line107 is to concat sequence along axis 0 (steps dimension) for each step. When direction is 'both', the **nextHidden** of above computation of each step are like [hf_1, hb_T], [hf_2, hb_T-1], ..., [hf_T, hb_1], using above proposal solution we will get the sequence [hf_T, hb_1, ..., hf_2, hb_T-1, hf_1, hb_T] which is not as expected [hf_1, hb_1, ..., hf_T, hb_T]. Is my understanding correct?

Reply:
I prefer to reversing the concatenation in the algorithm steps than "fix" the wrong order after that. 

> When direction is 'both', the nextHidden of above computation of each step are like [hf_1, hb_T], [hf_2, hb_T-1], ..., [hf_T, hb_1], using above proposal solution we will get the sequence [hf_T, hb_1, ..., hf_2, hb_T-1, hf_1, hb_T] which is not as expected [hf_1, hb_1, ..., hf_T, hb_T]. Is my understanding correct?

You are correct. To make "both" direction working, we probably need to save the forward and backward nextHidden separately and concatenate them in different order.